pkg/network/ovn_kubernetes.go

```diff
@@ -391,34 +405,76 @@ func bootstrapOVNHyperShiftConfig(hc HyperShiftConfig, kubeClient cnoclient.Clie
 	if !hc.Enabled {
 		return ovnHypershiftResult, nil
 	}
-	route := &routev1.Route{}
-	gvr := schema.GroupVersionResource{
-		Group:    "route.openshift.io",
-		Version:  "v1",
-		Resource: "routes",
-	}
-	clusterClient := kubeClient.ClientFor(MANAGEMENT_CLUSTER_NAME)
-	routeObj, err := clusterClient.Dynamic().Resource(gvr).Namespace(hc.Namespace).Get(context.TODO(), "ovnkube-sbdb", metav1.GetOptions{})
+
+	hcp := &hyperv1.HostedControlPlane{}
+	err := kubeClient.ClientFor(MANAGEMENT_CLUSTER_NAME).CRClient().Get(context.TODO(), types.NamespacedName{
+		Name:      hc.Name,
+		Namespace: hc.Namespace,
+	}, hcp)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.Infof("Did not find ovnkube-sbdb route")
-		} else {
-			return nil, fmt.Errorf("Could not get ovnkube-sbdb route: %v", err)
-		}
-	} else {
-		err := runtime.DefaultUnstructuredConverter.FromUnstructured(routeObj.UnstructuredContent(), route)
-		if err != nil {
-			return ovnHypershiftResult, err
+		return nil, fmt.Errorf("failed to retrieve ingress controller config, err: %v", err)
+	}
+	for _, svc := range hcp.Spec.Services {
+		// TODO: instead of the hardcoded string use ServiceType hyperv1.OVNSbDb once the API is updated
+		if svc.Service == "OVNSbDb" {
+			ovnHypershiftResult.ServicePublishingStrategy = &svc.ServicePublishingStrategy
 		}
-		if route.Spec.Host == "" && route.Status.Ingress[0].Host == "" {
-			return ovnHypershiftResult, nil
+	}
+	if ovnHypershiftResult.ServicePublishingStrategy == nil {
+		return nil, fmt.Errorf("service publishing strategy for OVN southband database does not exist in hyperv1.HostedControlPlane %s/%s", hc.Name, hc.Namespace)
```

Comment:
Should we generate a reasonable default if this is missing? Presumably we can just default to Route.